File: validation/complexity_g2b_20260812/reductions/recv.py
class Holder:
    def bare_self(self, n):
        return Holder.bare_self(self, n)          # type-qualified: expected 1

    def recv_self(self, n):
        return self.recv_self(n)                  # self-qualified: expected 1

    def same_name(self, n, other):
        return other.same_name(n)                 # other receiver: expected 0


def free_self(n):
    return free_self(n)                           # bare: expected 1

File: validation/complexity_g2b_20260812/reductions/recv.go
package recv

type holder struct{}
type other struct{}

func (o other) SameName(n int) int { return n }

// Bare self-name call. Expected 1.
func BareSelf(n int) int { return BareSelf(n) }

// Receiver-qualified self-call. Expected 1.
func (h holder) RecvSelf(n int) int { return h.RecvSelf(n) }

// Same member NAME, another receiver. Expected 0.
func (h holder) SameName(n int) int {
	var o other
	return o.SameName(n)
}
